src/components/stack-chart/Canvas.js

```diff
@@ -425,44 +460,60 @@ class StackChartCanvas extends React.PureComponent<Props> {
       return;
     }
     const { depth, stackTimingIndex } = hoveredItem;
-    const { stackTimingByDepth, updatePreviewSelection } = this.props;
+    const { combinedTimingRows, updatePreviewSelection } = this.props;
     updatePreviewSelection({
       hasSelection: true,
       isModifying: false,
-      selectionStart: stackTimingByDepth[depth].start[stackTimingIndex],
-      selectionEnd: stackTimingByDepth[depth].end[stackTimingIndex],
+      selectionStart: combinedTimingRows[depth].start[stackTimingIndex],
+      selectionEnd: combinedTimingRows[depth].end[stackTimingIndex],
     });
   };
 
-  _getCallNodeIndexFromHoveredItem(
+  _getCallNodeIndexOrMarkerIndexFromHoveredItem(
     hoveredItem: HoveredStackTiming | null
-  ): IndexIntoCallNodeTable | null {
+  ): {| index: number, type: 'marker' | 'call-node' |} | null {
     if (hoveredItem === null) {
       return null;
     }
 
     const { depth, stackTimingIndex } = hoveredItem;
-    const { stackTimingByDepth } = this.props;
-    const callNodeIndex = stackTimingByDepth[depth].callNode[stackTimingIndex];
-    return callNodeIndex;
+    const { combinedTimingRows } = this.props;
+
+    if (combinedTimingRows[depth].callNode) {
+      const callNodeIndex =
+        combinedTimingRows[depth].callNode[stackTimingIndex];
+      return { index: callNodeIndex, type: 'call-node' };
+    }
+
+    const index = combinedTimingRows[depth].index[stackTimingIndex];
+    return { index, type: 'marker' };
   }
+
   _onSelectItem = (hoveredItem: HoveredStackTiming | null) => {
     // Change our selection to the hovered item, or deselect (with
     // null) if there's nothing hovered.
-    const callNodeIndex = this._getCallNodeIndexFromHoveredItem(hoveredItem);
-    this.props.onSelectionChange(callNodeIndex);
+    const result = this._getCallNodeIndexOrMarkerIndexFromHoveredItem(
+      hoveredItem
+    );
+    if (result && result.type === 'call-node') {
```

Comment:
Please add a comment here explaining that marker selection is not currently implemented and refer to issue #2355

Reply:
👍